recipes/newrelic/infrastructure/otel/rhel.yml

```diff
@@ -0,0 +1,262 @@
+        - task: assert_pre_req
+        - task: install_otel_collector
+        - task: update_license_key
+        - task: update_otel_end_point
+        - task: restart_otel_service
+        - task: assert_otel_collector_status_ok
+        - task: post_install
+
+    write_recipe_metadata:
+      cmds:
+        - |
+          echo '{"Metadata":{"CapturedCliOutput":"true"}}' | tee {{.NR_CLI_OUTPUT}} > /dev/null
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_ROOT_OR_SUDO=$([ ! -z "$SUDO_USER" ] || [ $(whoami) = "root" ] && echo "true" || echo "false") # Assert root or sudo user
+          if [ "$IS_ROOT_OR_SUDO" = "false" ]; then
+            echo "This newrelic install must be run under sudo or root" >&2
+            exit 3
+          fi      
+        - |
+          IS_GREP_INSTALLED=$(which grep | wc -l)
+          if [ $IS_GREP_INSTALLED -eq 0 ] ; then
+            echo "grep is required to run the newrelic install. Please install grep and re-run the installation." >&2
+            exit 10
+          fi
+        - |
+          IS_SED_INSTALLED=$(which sed | wc -l)
+          if [ $IS_SED_INSTALLED -eq 0 ] ; then
+            echo "sed is required to run the newrelic install. Please install sed and re-run the installation." >&2
+            exit 11
+          fi
+        - |
+          IS_AWK_INSTALLED=$(which awk | wc -l)
+          if [ $IS_AWK_INSTALLED -eq 0 ] ; then
+            echo "awk is required to run the newrelic install. Please install awk and re-run the installation." >&2
+            exit 12
+          fi
+        - |
+          IS_CAT_INSTALLED=$(which cat | wc -l)
+          if [ $IS_CAT_INSTALLED -eq 0 ] ; then
+            echo "cat is required to run the newrelic install. Please install cat and re-run the installation." >&2
+            exit 13
+          fi
+        - |
+          IS_TEE_INSTALLED=$(which tee | wc -l)
+          if [ $IS_TEE_INSTALLED -eq 0 ] ; then
+            echo "tee is required to run the newrelic install. Please install tee and re-run the installation." >&2
+            exit 14
+          fi
+        - |
+          IS_TOUCH_INSTALLED=$(which touch | wc -l)
+          if [ $IS_TOUCH_INSTALLED -eq 0 ] ; then
+            echo "touch is required to run the newrelic install. Please install touch and re-run the installation." >&2
+            exit 15
+          fi
+      vars:
+        DISTRO_VERSION:
+          sh: |
+            rpm -E %{rhel}
+
+    install_otel_collector:
+      cmds:
+        - |
+          if [[ {{.AMAZON_LINUX_VERSION}} != "2" && {{.AMAZON_LINUX_VERSION}} != "2022" && {{.AMAZON_LINUX_VERSION}} != "2023" ]] ; then
+            REPO_URL=$(echo -n "{{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/linux/yum/{{.REPO_DIR}}/{{.DISTRO_VERSION}}/{{.ARCH}}/newrelic-infra.repo")
+          else
+            REPO_URL=$(echo -n "{{.NEW_RELIC_DOWNLOAD_URL}}infrastructure_agent/linux/yum/{{.REPO_DIR}}/{{.AMAZON_LINUX_VERSION}}/{{.ARCH}}/newrelic-infra.repo")
```

Comment:
Would this work for Amazon Linux 2022? Pacakges for AL 2022 are not uploaded anymore: https://github.com/newrelic/opentelemetry-collector-releases/blob/main/distributions/nr-otel-collector/upload-schema-linux-rpm.yml#L36

Reply:
will update PR accordingly.